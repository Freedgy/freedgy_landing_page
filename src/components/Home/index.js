import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';
import useWindowSize from '../../../static/js/useWindowSize';

const styles = theme => ({
    home: {
        "&&": {
            background: 'linear-gradient(180deg, #006666 10%, #00a0a0 70%)',
            width: "100%",
            margin: 0,
            padding: '10%',
            display: "flex",
            flexDirection: "row",
            justifyContent: "space-around",
            alignItems: "center",
            '@media only screen and (max-width: 992px)': {
                paddingTop: '12%',
                paddingBottom: '0%'
            },
        }
    },
    homeImage: {
        "&&": {
            width: "35%",
            height: 'auto'
        }
    },
    homeInfos: {
        "&&": {
            flex: "0 0 50%",
            maxWidht: '50%',
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
            '@media only screen and (max-width: 992px)': {
                flex: "0 0 90%",
            },
        }
    },
    homeText: {
        "&&": {
            width: "100%",
            height: "100%",
            justifyContent: "space-between",
            display: "flex",
            flexDirection: "column",
            alignItems: "center",
        }
    },
    button: {
        "&&": {
            borderColor: '#fff',
            backgroundColor: 'white',
            color: "#008080",
            size: "large",
            margin: "10%",
            '&:hover': {
                borderColor: 'white',
                color: "white",
                backgroundColor: 'transparent',
            },
        }
    }
});

function Home({ classes, datas }) {
    const size = useWindowSize();

    return (
        <section id="home" className={classes.home}>
            <Box className={classes.homeInfos}>
                <Box className={classes.homeText}>
                    <Typography variant="h1" color='secondary' align="center">
                        {datas.title}
                    </Typography>
                    <br />
                    <Typography variant="body1" color="secondary" align="justify">
                        {datas.description}
                    </Typography>
                </Box>
                <Button className={classes.button} variant="outlined" href="http://app.freedgy.com">
                    {datas.button}
                </Button>
            </Box>
            {size.width > 992 ? (<img alt="logo_freedgy" className={classes.homeImage} src={datas.image}></img>) : (null)}
        </section>
    );
}

Home.propTypes = {
    classes: PropTypes.object.isRequired,
    datas: PropTypes.object.isRequired,
};

export default withStyles(styles)(Home);