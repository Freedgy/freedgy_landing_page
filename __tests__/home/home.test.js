import React from "react"
import renderer from "react-test-renderer"

import Home from "../../src/components/Home"

import jsdom from 'jsdom';
const {JSDOM} = jsdom;
const {document} = (new JSDOM('<!doctype html><html><body></body></html>')).window;
global.document = document;
global.window = document.defaultView;
global.Image = window.Image;

describe("Home", () => {
    it("renders correctly", async () => {
        const datas = require("../../static/pages/pages.en.json");

        const tree = renderer
            .create(<Home lang="en" datas={datas.content.home} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})