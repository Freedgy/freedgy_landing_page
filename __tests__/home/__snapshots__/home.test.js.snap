// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<section
  className="Home-home-1"
  id="home"
>
  <div
    className="MuiBox-root MuiBox-root-6 Home-homeInfos-3"
  >
    <div
      className="MuiBox-root MuiBox-root-7 Home-homeText-4"
    >
      <h1
        className="MuiTypography-root MuiTypography-h1 MuiTypography-colorSecondary MuiTypography-alignCenter"
      >
        Welcome
      </h1>
      <br />
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorSecondary MuiTypography-alignJustify"
      >
        Every day, many people cannot afford to buy the food necessary to have a decent diet. Contradictorily, on average, each supermarket (in France) throws away 20 kilograms of food daily. Our goal is to redistribute that food to people in need such as students, homeless people, and big families by providing connected fridges to supermarkets.
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined Home-button-5"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        get started
      </span>
    </button>
  </div>
</section>
`;
